ops/demo/api.tf

```
@@ -14,7 +14,7 @@ module "simple_report_api" {
   app_settings = {
     SPRING_PROFILES_ACTIVE                = "azure-demo"
     SPRING_DATASOURCE_URL                 = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.sr_db_jdbc.id})"
-    APPLICATIONINSIGHTS_CONNECTION_STRING = "InstrumentationKey=${data.azurerm_application_insights.app_insights.instrumentation_key};IngestionEndpoint=https://eastus-1.in.applicationinsights.azure.com/"
+    APPLICATIONINSIGHTS_CONNECTION_STRING = "InstrumentationKey=${data.azurerm_application_insights.app_insights.instrumentation_key};IngestionEndpoint=${data.azurerm_application_insights.app_insights.connection_string}"
```

From viewing the application insights at https://portal.azure.com/#@cdc.onmicrosoft.com/resource/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-simple-report-prod/providers/Microsoft.Insights/components/prime-simple-report-prod-insights/overview I believe this should be

```suggestion
    APPLICATIONINSIGHTS_CONNECTION_STRING = ${data.azurerm_application_insights.app_insights.connection_string}
```

---

Hmm. According to https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net, the `InstrumentationKey` is a required field in the connection string.

---

interesting. It would be good to run Terraform apply in test and make sure it works

---

Short answer: it looks like that doesn't work, due to the format of the actual resource output. Working on the fix.

---

Updated. It looks like the output from that data source is actually the entire connection string (in the `InstrumentationKey=x;IngestionEndpoint=y` format), so we can just use that variable as-is.

---

To clarify, the above was tested in test and we're seeing data flowing in again.